app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/login/LoginScreen.kt

```diff
@@ -47,47 +55,123 @@ fun LoginScreen(
                 signedInState = signedInState,
                 messageBarState = messageBarState,
                 onButtonClicked = {
-                    loginViewModel.saveSignedInState(signedIn = true)
-                },
+                    loginViewModel.oneTapSignIn()
+                }
             )
-        },
+        }
+    )
+
+    val launcher = rememberLauncherForActivityResult(
+        ActivityResultContracts.StartIntentSenderForResult()
+    ) { result ->
+        try {
+            if (result.resultCode == Activity.RESULT_OK) {
+                loginViewModel.verifyToken(intent = result.data)
+            } else {
+                loginViewModel.saveSignedInState(signedIn = false)
+            }
+        } catch (e: ApiException) {
+            loginViewModel.saveSignedInState(signedIn = false)
+        }
+    }
+
+    fun launch(signInResult: BeginSignInResult) {
+        val intent = IntentSenderRequest.Builder(signInResult.pendingIntent.intentSender).build()
+        launcher.launch(intent)
+    }
+
+    OneTapSignIn(
+        launch = {
+            launch(it)
+        }
     )
+
+    LaunchedEffect(key1 = apiResponse) {
```

Comment:
Can we move some of this to the ViewModel? So instead of having the Composable handle success/error etc., we get the ViewModel to do this. And we expose some kind of ViewState for the Composable to react to that will trigger the `navigateToHomeScreen`?

Reply:
Refactored and comitted changes